### PR TITLE
Drop Python 3.3 support

### LINKED ISSUE #603: Drop support for Python 3.3
gcloud-python, django, and many libraries have both dropped support for 3.3. Is there anything preventing us from doing the same?

@thobrla ?

### PR DESCRIPTION
**Pull request by dhermes**
Also remove some trailing whitespace in `tox.ini` (my editor did this) and fixing up some double quote / single quote mixing in `setup.py`.

Fixes #603.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -128,8 +128,8 @@ Running Tests
     more information.
 
 -   **Note** that `django` related tests are turned off for Python 2.6
```
Comment (nathanielmanistaatgoogle) [inline CONTRIBUTING.md]: We no longer support 2.6.

File: docs/index.rst [hunk 2]
```diff
@@ -107,13 +107,13 @@ contributor license agreement.
...
+We support Python 2.7 and 3.4+. (Whatever this file says, the truth is
 always represented by our `tox.ini`_).
 
 .. _tox.ini: https://github.com/google/oauth2client/blob/master/tox.ini
 
 We explicitly decided to support Python 3 beginning with version
-3.3. Reasons for this include:
+3.4. Reasons for this include:
 
 * Encouraging use of newest versions of Python 3
```
Comment (nathanielmanistaatgoogle) [inline docs/index.rst:118]: Any desire to mention Django in a new item in this list?

File: setup.py [hunk 3]
```diff
@@ -41,27 +41,27 @@
     'six>=1.6.1',
 ]
 
-long_desc = """The oauth2client is a client library for OAuth 2.0."""
```
Comment (nathanielmanistaatgoogle) [inline setup.py:44]: Drop "The" from this text fragment?

**Comment by nathanielmanistaatgoogle** [comment]
Looks mostly good.

File: CONTRIBUTING.md (hunk 1 above)
Comment (dhermes) [inline CONTRIBUTING.md]: Nice. Good call.

File: setup.py (hunk 3 above)
Comment (dhermes) [inline setup.py:44]: Right right

File: docs/index.rst (hunk 2 above)
Comment (dhermes) [inline docs/index.rst:118]: Yes. Did it.

**Comment by dhermes** [comment]
@nathanielmanistaatgoogle PTAL. After the LGTM please remind me to squash.

**Comment by nathanielmanistaatgoogle** [comment]
Looks great. Squash! And merge.

**Comment by jonparrott** [comment]
Thank you @dhermes!

**Pull request merged**